### DISCUSSION TITLE
Object modes to be RX, not RWX

### DISCUSSION THREAD

**Discussion by scholarsmate** [ctc-oss/fapolicy-analyzer#292] [Developers]
fanotify, and by extension, fapolicyd, only cares about if a file is being opened for reading, executing or either reading or executing (any).  It does not consider write mode.  So the possible states are:

Allow **R**ead
Allow e**X**ecute
Allow _any_  **R**ead or e**X**excute

**Comment by tparchambault** [comment]
IMO, from a user's perspective, i.e. the man (or sysadmin) on the street's perspective, I believe the RX convention will not be obvious. It would require a bit of tribal knowledge of fapolicyd to know that writes are not considered. Possibly RWX with the 'W' in light gray or some other low contrast color to function as a placeholder, and a note at the bottom to the pane to communicate that write operations are not analyzed/considered by fapolicyd.

**Comment by scholarsmate** [comment]
@tparchambault, that's an interesting idea.  I've converted this issue over to a discussion, so it can be debated a bit more.

**Reply by tparchambault** [reply]
@scholarsmate I brought up the convention because the familiar "RWX" grouping by itself implies a context to the intended user demographic for quickly grokking what the "Mode" column represents. If 'W' is always permitted (which it is) then using the traditional 'RWX' triplet can be one less thing to interpret in a sea (!) of new symbolic conventions, e.g. "ST/AT/U", "A/D" even with the non-varying "W." Or at minimum, provide a dash in the write field, and pop-up help in the column header to explain the situation.

**Comment by jw3** [comment]
This is now mapped directly to the fapolicyd perm

- https://github.com/ctc-oss/fapolicy-analyzer/pull/742
